### PR TITLE
Code for Improved DSR GAN

### PR DESCRIPTION
**Pull request by harveymannering**
I have also included code for finetuning here which is currently not completely working.  We can maybe remove this if we would prefer to have all the code in this repo fully functional.  

No need for a full review here, just let me know whether or not you want the finetuning code to be included.

### CODE REVIEW

**Comment by mxochicale** [comment]
Hi @harveymannering 

This looks great, the pre-trained models in huggingface are amazing. It would be great that you add in the README further description on which order and refer to https://github.com/budai4medtech/xfetus/tree/harvey/improved-dsr-gan/examples/myriad-guidelines when it is needed.

Thanks, Miguel

File: xfetus/improved-dsrgan/finetune_ddpm.py [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+    ##################
+    ##   1. SETUP   ##
+    ##################
+
+    # Command line aurgments - for script
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-d", "--dataset_path", help="File location of the fetal brain dataset", type=str)
+    parser.add_argument("-w", "--wandb_enabled", help="Enable weights and bias logging", type=bool)
+    parser.add_argument("-c", "--ddpm_checkpoint_path", default=None,  help="File location for the pre-trained DDPM model", type=str)
+    parser.add_argument("-l", "--ddpm_pa_loss", default=None,  help="Use DDPM-PA loss (as opposed to standard DDPM loss)", type=bool)
```
Comment (harveymannering) [inline xfetus/improved-dsrgan/finetune_ddpm.py:31]: DDPM-PA Loss is probably no longer needed.  We can keep in the code for now since it is an optional argument, but if we refactor, DDPM-PA should be removed.

File: xfetus/improved-dsrgan/loss.py
Comment (harveymannering) [inline xfetus/improved-dsrgan/loss.py:1]: Again, DDPM-PA is probably superfluous now.  I experimented with this loss, but ultimately it didn't work.  If we refactor the code, we can remove most of this file.
